File: 20210523LangReview/Swift/02-ctor/ctor.py
"""
[ expriment result ]
===
- it turns out that Python suppurts ctor inheritance
===
"""

class Base:
    def __init__(self, x: int):
        self.x = x

class Derived(Base):
    pass

if __name__ == "__main__":
    d1: Derived = Derived(1)
    print(d1.x)

File: 20210523LangReview/Swift/02-ctor/ctor.js
/*

[ expriment result ]
===
- it turns out that JavaScript suppurts ctor inheritance
===
*/ 

class Base {
    m_x;
    constructor(x) {
        this.m_x = x;
    }
}

class Derived extends Base {

}

let d1 = new Derived(1); // OK
console.log(`${d1.m_x}`);